libsoftwarecontainer/src/gateway/networkgateway.cpp

```diff
@@ -347,3 +333,54 @@ bool NetworkGateway::isBridgeAvailable()
     return isSuccess(m_netlinkHost.hasAddress(addresses, AF_INET, m_gateway.c_str()));
 }
 
+std::string IPTableEntry::getChain() {
+    if ( "INCOMING" == m_type) {
+        return "INPUT";
+    } else if ("OUTGOING" == m_type) {
+        return "OUTPUT";
+    } else if ("FORWARD" == m_type) {
+        return "FORWARD";
+    }
+    return "";
+}
+
+bool IPTableEntry::applyRules()
+{
+    for (auto rule:m_rules) {
+        if (!insertRule(rule, getChain())) {
+            log_error() << "Couldn't apply the rule " << rule.target;
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool IPTableEntry::insertRule(Rule rule, std::string type)
+{
+    if (type.empty()) {
+        log_error() << "Chain type is not recognized :" << type;
+        return false;
+    }
+
+    std::string iptableCommand = "iptables -A " + type;
+
+    if (!rule.host.empty()) {
+        iptableCommand = iptableCommand + " -s " + rule.host ;
+    }
+
+    if (rule.ports.any) {
+        if (rule.ports.multiport) {
+            iptableCommand = iptableCommand + " -p tcp --match multiport --sports " + rule.ports.ports;
+        } else {
+            iptableCommand = iptableCommand + " -p tcp --sport " + rule.ports.ports;
+        }
+
+    }
+
+    iptableCommand = iptableCommand + " -j " + rule.target;
+    log_debug() << "Add network rule : " <<  iptableCommand;
+
+    system(iptableCommand.c_str());
```

Comment:
Check the return code here. Also, I think we should try to use the glibmm functions available instead of system(), such as spawn_sync()

https://developer.gnome.org/glibmm/stable/group__Spawn.html